cinder/scheduler/filter_scheduler.py

```diff
@@ -109,6 +109,43 @@ def schedule_create_volume(self, context, request_spec, filter_properties):
                                          filter_properties,
                                          allow_reschedule=True)
 
+    def find_backend_for_connector(self, context, connector, request_spec):
+        key = 'connection_capabilities'
+        if key not in connector:
+            raise exception.InvalidConnectionCapabilities(
+                reason=_("The connector doesn't contain a %s field.") % key)
+
+        weighed_backends = self._get_weighted_candidates(context, request_spec)
+        if not weighed_backends:
+            raise exception.NoValidBackend(reason=_("No weighed backends "
+                                                    "available"))
+        connector_capabilities = set(connector[key])
+
+        def _backend_matches_connector(bck):
+            if key not in bck.obj.capabilities:
+                LOG.debug("Backend %(backend) doesn't contain %(key)s.",
+                          {'backend': bck.obj.host, 'key': key})
+                return False
+            backend_capabilities = set(bck.obj.capabilities.get(key))
+            if connector_capabilities & backend_capabilities ==\
+                    connector_capabilities:
+                return True
+            LOG.debug("Requested %(key)s %(req)s not found in "
+                      "%(host)s backend's %(key)s %(given)s.",
+                      {'key': key, 'req': connector_capabilities,
+                       'host': bck.obj.host, 'given': backend_capabilities})
+            return False
+
+        weighed_backends = [b for b in weighed_backends if
+                            _backend_matches_connector(b)]
+
+        if not weighed_backends:
+            raise exception.NoValidBackend(
+                reason=_("No backend matched the given connector."))
+
+        top_backend = self._choose_top_backend(weighed_backends, request_spec)
+        return top_backend.obj
+
```

Comment:
Can this be added as a new scheduler filter so we don't have to merge these changes when upgrading to Train?

Reply:
I don't think we can just create a new filter for all of this, since filtering/scheduling is not something that is done during attachments.
Or I didn't get your question well enough? Do you mean a new filter, or a new filter_scheduler driver ?

Reply:
My best guess would be, that he wants you to use the functions already there and write an additional filter for the scheduler, that only reacts to a certain attribute in the request spec or something. Nova does that for rebuilds iirc.

Reply:
I guess I was thinking of a derived class that only had our changes to it, so it would be trivial to upgrade cinder versions.  If it's not possible it's ok.  I'm just always thinking about the issues of diffs between upstream and our hacks to make our stuff work.